File: 4.js



function matchChar(aray) {
    
    if(!Array.isArray(aray)) {
        throw "Parameter harus Array"
    }

    let txt = aray.toString(aray);
     for (const re of aray) {
         
        let regex = new RegExp('[' + re + ']' + "{3,}", "g");
        let hasil = regex.exec(txt);
        
        console.log(hasil[0]);
        
     }
    //  console.log(txt);
     
    
}

let theAray = ['cat', 'listen', 'code', 'act', 'silent', 'tac']

let test = /[cat]{3,}/
let str = theAray.toString();


let hasil = test.matchAll(str);
for (const match of hasil) {

    console.log(match);
    console.log(match.index);
    
}

// matchChar(theAray)

File: 4.py
import re



def findSame(array) :

    if(isinstance(array, list) is not True) :
        print("Parameter Harus array")
        return

    txt = ",".join(array)
    hasil = ""

    for i in array :

        test = r"[" + re.escape(i) + "]{3,}"
        regex = re.findall(test, txt)

        if(len(regex) is 1) :
            continue


        if(regex[0] != hasil) :

            print(hasil + "-"+ regex[0])
            hasil = regex[0]
            # print(regex)








theAray = ['cat', 'listen', 'code', 'act', 'silent', 'tac']

txt = "aku kamu"

findSame(theAray)
    